### PR TITLE
Add more consequences to SO_TERMS

### PR DESCRIPTION
**Pull request by maehler**
This PR adds a functionality or fixes a bug.

There are consequences (SO terms) [listed in the VEP documentation](https://www.ensembl.org/info/genome/variation/prediction/predicted_data.html) that are not in `SO_TERMS`. This pull request adds them. The severity seems to match with what VEP lists, so I just added these accordingly.

**How to test**:
1. Try to load a case with that has variants whose consequence is either `splice_donor_5th_base_variant`, `splice_donor_region_variant` or `splice_polypyrimidine_tract_variant`.
2. See the error message below.

    <details>
    <summary>Error message</summary>

    ```
    2022-08-18 16:29:03 vs842 scout.adapter.mongo.variant_loader[1966] INFO Start inserting clinical cancer variants into database
    2022-08-18 16:29:03 vs842 scout.adapter.mongo.variant_loader[1966] ERROR unexpected error
    Traceback (most recent call last):
      File "/scout_prod/scout/scout/adapter/mongo/variant_loader.py", line 701, in load_variants
        nr_inserted = self._load_variants(
      File "/scout_prod/scout/scout/adapter/mongo/variant_loader.py", line 432, in _load_variants
        parsed_variant = parse_variant(
      File "/scout_prod/scout/scout/parse/variant/variant.py", line 285, in parse_variant
        for parsed_transcript in parse_transcripts(raw_transcripts):
      File "/scout_prod/scout/scout/parse/variant/transcript.py", line 80, in parse_transcripts
        transcript["region_annotations"] = get_regional_annotation(functional_annotations)
      File "/scout_prod/scout/scout/parse/variant/transcript.py", line 292, in get_regional_annotation
        regional_list.append(SO_TERMS[annotation]["region"])
    KeyError: 'splice_polypyrimidine_tract_variant'
    ```
    </details>

3. Do `scout view cases` and see that the case is still added.

**Expected outcome**:
Variants should be loaded without errors.

**Review:**
- [x] code approved by DN
- [x] tests executed by automatic tests.

### CODE REVIEW

**Comment by dnil** [comment]
Thanks! E.g. splice_donor_5th_base_variant is indeed in the SO (https://rgd.mcw.edu/rgdweb/ontology/view.html?acc_id=SO:0001787&offset=27) nowadays, but not listed in the constants. Perhaps we should dl and parse the full SO really. Anyway, very useful!

File: scout/constants/so_terms.py [hunk 1]
```diff
@@ -15,30 +15,33 @@
...
-    "TFBS_amplification": {"rank": 33, "region": "TFBS"},
-    "TF_binding_site_variant": {"rank": 34, "region": "TFBS"},
-    "regulatory_region_amplification": {"rank": 35, "region": "regulatory_region"},
-    "regulatory_region_variant": {"rank": 36, "region": "regulatory_region"},
-    "feature_elongation": {"rank": 37, "region": "genomic_feature"},
-    "feature_truncation": {"rank": 38, "region": "genomic_feature"},
-    "intergenic_variant": {"rank": 39, "region": "intergenic_variant"},
+    "splice_donor_5th_base_variant": {"rank": 16, "region": "splicing"},
+    "splice_donor_region_variant": {"rank": 17, "region": "splicing"},
+    "splice_polypyrimidine_tract_variant": {"rank": 18, "region": "splicing"},
```
Comment (northwestwitch) [inline scout/constants/so_terms.py:20]: Looks good to me. Any reason why we should merge it as it is @dnil?

Comment (dnil) [reply scout/constants/so_terms.py:20]: I can check, but I dont think so! It was a draft when I looked last. 😊

**Review by dnil — APPROVED** [review]
Looks great!

File: scout/constants/so_terms.py (hunk 1 above)
Comment (dnil) [reply scout/constants/so_terms.py:20]: Ah, and there was the idea to parse directly from the full SO or retrieve from VEP, but that is likely overkill.

Comment (dnil) [reply scout/constants/so_terms.py:20]: Do you have a case with those terms mixed in @northwestwitch for testing?

Comment (dnil) [reply scout/constants/so_terms.py:20]: Otw the regular tests pass, and this shoould work..

Comment (northwestwitch) [reply scout/constants/so_terms.py:20]: > Do you have a case with those terms mixed in @northwestwitch for testing?

Nope, unfortunately

Comment (northwestwitch) [reply scout/constants/so_terms.py:20]: > Otw the regular tests pass, and this shoould work..

Agreed!

**Review by northwestwitch — APPROVED** [review]
👍🏻

**Comment by northwestwitch** [comment]
Hey @maehler did you already test this branch when loading one or more of your samples?

**Comment by maehler** [comment]
Yes, I loaded a handful of samples without any apparent issues.

**Comment by northwestwitch** [comment]
Thanks! So we don't have to test another time here in Stockholm

**Pull request merged**